docs/guide/migration.md

```diff
@@ -27,27 +27,39 @@ A small fraction of users will now require using [@vitejs/plugin-legacy](https:/
 
-## Dev Server Changes
+## Optimized deps and SSR changes
+
+These are the biggest architecture changes in Vite v3. To allow projects to migrate from v2 in case of a compat issue, options have been added to revert to the Vite v2 strategies.
+
+:::warning
+These options are marked as experimental and deprecated. They may be removed in a future v3 minor.
+
+- `experimental.devDepsScanner`
+- `experimental.buildRollupPluginCommonjs`
+- `experimental.buildSsrCjsExternalHeuristics`
+  :::
+
+### Dev Server Changes
 
 Vite's default dev server port is now 5173. You can use [`server.port`](../config/server-options.md#server-port) to set it to 3000.
 
 Vite's default dev server host is now `localhost`. You can use [`server.host`](../config/server-options.md#server-host) to set it to `127.0.0.1`.
 
 Vite optimizes dependencies with esbuild to both convert CJS-only deps to ESM and to reduce the number of modules the browser needs to request. In v3, the default strategy to discover and batch dependencies has changed. Vite no longer pre-scans user code with esbuild to get an initial list of dependencies on cold start. Instead, it delays the first dependency optimization run until every imported user module on load is processed.
 
-To get back the v2 strategy, you can use [`optimizeDeps.devScan`](../config/dep-optimization-options.md#optimizedepsdevscan).
+To get back the v2 strategy, you can use [`experimental.devDepsScanner`](../config/dep-optimization-options.md#optimizedepsdevscan).
 
-## Build Changes
+### Build Changes
 
 In v3, Vite uses esbuild to optimize dependencies by default. Doing so, it removes one of the most significant differences between dev and prod present in v2. Because esbuild converts CJS-only dependencies to ESM, [`@rollupjs/plugin-commonjs`](https://github.com/rollup/plugins/tree/master/packages/commonjs) is no longer used.
 
-If you need to get back to the v2 strategy, you can use [`optimizeDeps.disabled: 'build'`](../config/dep-optimization-options.md#optimizedepsdisabled).
+If you need to get back to the v2 strategy, you can use [`experimental.buildRollupPluginCommonjs: true`](../config/dep-optimization-options.md#experimental-buildrollupplugincommonjs).
 
-## SSR Changes
+### SSR Changes
 
 Vite v3 uses ESM for the SSR build by default. When using ESM, the [SSR externalization heuristics](https://vitejs.dev/guide/ssr.html#ssr-externals) are no longer needed. By default, all dependencies are externalized. You can use [`ssr.noExternal`](../config/ssr-options.md#ssrnoexternal) to control what dependencies to include in the SSR bundle.
 
-If using ESM for SSR isn't possible in your project, you can set `ssr.format: 'cjs'` to generate a CJS bundle. In this case, the same externalization strategy of Vite v2 will be used.
+If using ESM for SSR isn't possible in your project, you can set `experimental.buildSsrCjsExternalHeuristics: true` to generate a CJS bundle using the same externalization strategy of Vite v2.
```

Comment:
We could add a link here too

Reply:
Let's first discuss this in the meeting before further documenting the options.
Since we would like to be able to remove them, maybe the docs could only be in the migration guide?

Reply:
I think that works too 👍